src/lighteval/models/base_model.py

```diff
@@ -345,6 +353,145 @@ def greedy_until_with_logits(
             override_bs=override_bs,
         )
 
+    def batched(self, iterable, n):
```

Comment:
What is this function for (compared to the usual Batch creation?)

Reply:
This was needed to batch results two by two. i.e [1, 2, 3, 4] -> [1,2] [3, 4].
There is a function doing this in the standard lib but only accessible for python 3.12.
Should we force python 3.12 and above ?

Reply:
Otherwise, I can put this function in utils.

Reply:
In utils would be fine, let's not force Python 3.12